lib/Auth/Basic.php

```diff
@@ -340,24 +424,34 @@ public function check()
             $this->debug('User is already authenticated');
         }
     }
-    /** Add additional info to be stored in user session. */
+
+    /**
+     * Add additional info to be stored in user session.
+     *
+     * @param string|array $key
+     * @param mixed $val
+     *
+     * @return $this;
+     */
     public function addInfo($key, $val = null)
     {
         if (is_null($val) && is_array($key)) {
             foreach ($key as $a => $b) {
                 $this->addInfo($a, $b);
             }
-
-            return;
```

Comment:
we need return $this; here but overall, I prefer returning from if() blocks early. It flatters the rest of the logic and keeps it more readable. It's not a big deal here, but if you look at a complex methods, it's always using return as early as it can.